src/main/java/com/conveyal/datatools/manager/controllers/api/UserController.java

```diff
@@ -145,79 +141,158 @@ private static int getUserCount(Request req, Response res) {
+         * This will prevent them from being overwritten when creating permissions for other datatools instances.
+         */
+        String existingPermissions = "";
 
-        return executeRequestAndGetResult(createUserRequest, req);
+        /**
+         * Used to hold email, userId and expanded datatools parameters if they match the AUTH0_CLIENT_ID parameter.
+         */
+        Auth0UserProfile user;
     }
 
+    /**
+     * Update user permissions. Only used in cases where the Auth0 user already exists and permissions for the current
+     * instance of datatools has not been set previously.
+     */
+//    private static String updateUserPermissions(Request req, Auth0UserSearchResult searchResult, String newPermissions) {
+//        // if there are existing permissions the assumption is that the datatools info has been set, but, only if it
+//        // matches the current instance of datatools (AUTH0_CLIENT_ID).
+//        if(!searchResult.existingPermissions.isEmpty() && searchResult.user.getApp_metadata().getDatatoolsInfo() != null) {
+//            LOG.info("Permissions for this Auth0 user {} already exist for this instance {} of datatools",
+//                searchResult.user.getEmail(), AUTH0_CLIENT_ID);
+//            logMessageAndHalt(
+//                req,
+//                400,
+//                String.format("User %s already exists", searchResult.user.getEmail())
+//            );
+//        }
+//
+//        LOG.info("Auth0 user {} already exists, updating permissions", searchResult.user.getEmail());
+//        // User is unknown to this instance (AUTH0_CLIENT_ID) of datatools, update permissions, making sure to
+//        // preserve any previously held permissions.
+//        String permissions = (searchResult.existingPermissions.isEmpty()) ?
+//            newPermissions : String.format("%s,%s", searchResult.existingPermissions, newPermissions);
+//        String json = String.format("{ \"app_metadata\": { \"datatools\" : [%s]  }}",permissions);
+//        HttpPatch updateUserRequest = new HttpPatch(getUserIdUrl(req, searchResult.user.getUser_id()));
+//        setHeaders(req, updateUserRequest);
+//        setRequestEntityUsingJson(updateUserRequest, json, req);
+//        return executeRequestAndGetResult(updateUserRequest, req);
+//    }
+
     private static String updateUser(Request req, Response res) {
         String userId = req.params("id");
         Auth0UserProfile user = getUserById(userId);
+        JsonNode jsonNode = JsonUtil.parseJsonFromBody(req);
+        JsonNode data = jsonNode.get("data");
+        String json = "{ \"app_metadata\": { \"datatools\" : " + data + " }}";
+        return updateUser(user, userId, json, req);
+    }
 
+    /**
+     * Perform an update user request with Auth0.
+     * @param user - the user to update
+     * @param json - the permissions (or other object) as a JSON string
+     * @param req - the original HTTP request
+     * @return the response string
```

Comment:
javadoc does not match (missing userId). Although, I'm not really sure that userId is necessary? Can't that be derived from user?

Reply:
@landonreed The userId is used in the logMessageAndHalt call when the user parameter is null.